### PR TITLE
Run nativeaot tests in helix

### PR DESCRIPTION
**Pull request by sbomer**
This changes the existing NativeAot smoke test runs to run in helix. I changed the multi-module test run (only supported on windows) to run only in the checked configuration, to avoid having to add a separate windows build script invocation inside one of the yaml templates.

### CODE REVIEW

**Comment by agocke** [comment]
Hm, not sure how this works. Weren't we already running the smoke tests on the build machine? Does this disable that run? Or just add a new one?

**Comment by agocke** [comment]
(nvm didn't notice this was a draft)

File: eng/pipelines/common/templates/runtimes/build-runtime-tests-and-send-to-helix.yml [hunk 1]
```diff
@@ -26,11 +27,11 @@ parameters:
 
 steps:
   - ${{ if eq(parameters.runtimeFlavor, 'mono') }}:
-    - script: $(Build.SourcesDirectory)/src/tests/build$(scriptExt) /p:RuntimeVariant=${{ parameters.runtimeVariant }} /p:LibrariesConfiguration=${{ parameters.buildConfig }} -mono -ci os ${{ parameters.osGroup }} ${{ parameters.archType }} $(buildConfigUpper)
+    - script: $(Build.SourcesDirectory)/src/tests/build$(scriptExt) /p:RuntimeVariant=${{ parameters.runtimeVariant }} ${{ parameters.testBuildArgs }} -mono os ${{ parameters.osGroup }} ${{ parameters.archType }} $(buildConfigUpper) /p:LibrariesConfiguration=${{ parameters.buildConfig }} -ci
       displayName: Build Tests
 
   - ${{ if ne(parameters.runtimeFlavor, 'mono') }}:
-    - script: $(Build.SourcesDirectory)/src/tests/build$(scriptExt) /p:LibrariesConfiguration=${{ parameters.buildConfig }} -ci os ${{ parameters.osGroup }} ${{ parameters.archType }} $(buildConfigUpper)
+    - script: $(Build.SourcesDirectory)/src/tests/build$(scriptExt) ${{ parameters.archType }} $(buildConfigUpper) ${{ parameters.testBuildArgs }} -ci
```
Comment (trylek) [inline eng/pipelines/common/templates/runtimes/build-runtime-tests-and-send-to-helix.yml:34]: Why don't we need the LibrariesConfiguration in this case?

**Review by trylek — APPROVED** [review]
LGTM, thank you!

File: eng/pipelines/runtime.yml [hunk 2]
```diff
@@ -240,17 +266,30 @@ extends:
...
+          variables:
+          - name: timeoutPerTestInMinutes
+            value: 60
+          - name: timeoutPerTestCollectionInMinutes
+            value: 180
           jobParameters:
             testGroup: innerloop
             timeoutInMinutes: 120
-            nameSuffix: NativeAOT
+            nameSuffix: NativeAOT_Helix
```
Comment (MichalStrehovsky) [inline eng/pipelines/runtime.yml:285]: Why do we need to change the suffix here?

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

**Pull request merged**